pkg/yurtctl/cmd/convert/convert.go

```diff
@@ -392,30 +425,31 @@ func (co *ConvertOptions) RunConvert() (err error) {
 	}
 
 	ctx = map[string]string{
-		"provider":              string(co.Provider),
-		"action":                "convert",
-		"yurtctl_servant_image": co.YurctlServantImage,
-		"yurthub_image":         co.YurhubImage,
-		"joinToken":             joinToken,
-		"kubeadm_conf_path":     co.KubeadmConfPath,
+		"action":             "convert",
+		"node_servant_image": co.NodeServantImage,
+		"yurthub_image":      co.YurhubImage,
+		"joinToken":          joinToken,
+		"kubeadm_conf_path":  co.KubeadmConfPath,
 	}
 
 	if co.YurthubHealthCheckTimeout != defaultYurthubHealthCheckTimeout {
 		ctx["yurthub_healthcheck_timeout"] = co.YurthubHealthCheckTimeout.String()
 	}
 
 	// 9. deploy yurt-hub and reset the kubelet service on edge nodes.
-	klog.Infof("deploying the yurt-hub and resetting the kubelet service on edge nodes...")
-	ctx["sub_command"] = "edgenode"
-	if err = kubeutil.RunServantJobs(co.clientSet, ctx, edgeNodeNames); err != nil {
-		klog.Errorf("fail to run ServantJobs: %s", err)
-		return
+	if len(edgeNodeNames) != 0 {
+		klog.Infof("deploying the yurt-hub and resetting the kubelet service on edge nodes...")
+		ctx["working_mode"] = "edge"
+		if err = kubeutil.RunServantJobs(co.clientSet, ctx, edgeNodeNames); err != nil {
+			klog.Errorf("fail to run ServantJobs: %s", err)
+			return
+		}
+		klog.Info("complete deploying yurt-hub on edge nodes")
 	}
-	klog.Info("complete deploying yurt-hub on edge nodes")
 
 	// 10. deploy yurt-hub and reset the kubelet service on cloud nodes
 	klog.Infof("deploying the yurt-hub and resetting the kubelet service on cloud nodes")
-	ctx["sub_command"] = "cloudnode"
+	ctx["working_mode"] = "cloud"
 	if err = kubeutil.RunServantJobs(co.clientSet, ctx, co.CloudNodes); err != nil {
 		klog.Errorf("fail to run ServantJobs: %s", err)
```

Comment:
The `klog.Error()`  seems a bit redundant,  because the returned error message will be printed by the caller. 
How about we remove those `klog.Error()` and just return the error message?   For examples: 
```
if err = kubeutil.RunServantJobs(co.clientSet, ctx, co.CloudNodes); err != nil {
    return fmt.Errorf("fail to run ServantJobs: %s", err)
}
```

Reply:
agree +1